test/common/upstream/health_checker_impl_test.cc

```diff
@@ -2598,6 +2517,18 @@ class TestProdHttpHealthChecker : public ProdHttpHealthCheckerImpl {
 
 class ProdHttpHealthCheckerTest : public HttpHealthCheckerImplTest {
 public:
+  void allocHealthChecker(envoy::config::core::v3::HealthCheck conf) {
+    health_checker_ = std::make_shared<TestProdHttpHealthChecker>(
+        *cluster_, conf, dispatcher_, runtime_, random_, HealthCheckEventLoggerPtr(event_logger_));
+  }
+
+  void addCompletionCallback() {
```

Comment:
ditto

Reply:
ping: this is identical to the version in the base class. I think you can drop this one.

Reply:
If removing this one then the unit test fails with "INFO: Build completed, 1 test FAILED, 5 total actions".
Logs 
`[2020-05-13 13:00:18.102][14][critical][backtrace] [bazel-out/k8-fastbuild/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #16: __libc_start_main [0x7f1bdb445830]
external/bazel_tools/tools/test/test-setup.sh: line 310:    14 Segmentation fault      (core dumped) "${TEST_PATH}" "$@" 2>&1`
 
   `

Reply:
Is it fine to make addCompletionCallback() and allocHealthChecker() to generic ?

Reply:
That could work but I would probably not templatize the parameter; this scenario IMO calls for polymorphism, if you want to keep the inheritance.

So just have addCompletionCallback without the param, and make virtual function to return the health-checker in the base class. Override the virtual function in the subclasses.